docs/book/component-guide/model-deployers/vertex.md

```diff
@@ -0,0 +1,179 @@
+* You need enterprise-grade model serving capabilities with autoscaling
+* You want a fully managed solution for hosting ML models
+* You need to handle high-throughput prediction requests
+* You want to deploy models with GPU acceleration
+* You need to monitor and track your model deployments
+
+This is particularly useful in the following scenarios:
+* Deploying models to production with high availability requirements
+* Serving models that need GPU acceleration
+* Handling varying prediction workloads with autoscaling
+* Integrating model serving with other GCP services
+
+{% hint style="warning" %}
+The Vertex AI Model Deployer requires a Vertex AI Model Registry to be present in your stack. Make sure you have configured both components properly.
+{% endhint %}
+
+## How to deploy it?
+
+The Vertex AI Model Deployer is provided by the GCP ZenML integration. First, install the integration:
+
+```shell
+zenml integration install gcp -y
+```
+
+### Authentication and Service Connector Configuration
+
+The Vertex AI Model Deployer requires proper GCP authentication. The recommended way to configure this is using the ZenML Service Connector functionality:
+
+```shell
+# Register the service connector with a service account key
+zenml service-connector register vertex_deployer_connector \
+    --type gcp \
+    --auth-method=service-account \
+    --project_id=<PROJECT_ID> \
+    --service_account_json=@vertex-deployer-sa.json \
+    --resource-type gcp-generic
+
+# Register the model deployer
+zenml model-deployer register vertex_deployer \
+    --flavor=vertex \
+    --location=us-central1
+
+# Connect the model deployer to the service connector
+zenml model-deployer connect vertex_deployer --connector vertex_deployer_connector
+```
+
+{% hint style="info" %}
+The service account needs the following permissions:
+- `Vertex AI User` role for deploying models
+- `Vertex AI Service Agent` role for managing model endpoints
+{% endhint %}
+
+## How to use it
+
+### Deploy a model in a pipeline
+
+Here's an example of how to use the Vertex AI Model Deployer in a ZenML pipeline:
+
+```python
+from typing_extensions import Annotated
+from zenml import ArtifactConfig, get_step_context, step
+from zenml.client import Client
+from zenml.integrations.gcp.services.vertex_deployment import (
+    VertexAIDeploymentConfig,
+    VertexDeploymentService,
+)
+
+@step(enable_cache=False)
+def model_deployer(
+    model_registry_uri: str,
+) -> Annotated[
+    VertexDeploymentService, 
+    ArtifactConfig(name="vertex_deployment", is_deployment_artifact=True)
+]:
+    """Model deployer step."""
+    zenml_client = Client()
+    current_model = get_step_context().model
+    model_deployer = zenml_client.active_stack.model_deployer
+
+    # Configure the deployment
+    vertex_deployment_config = VertexAIDeploymentConfig(
+        location="europe-west1",
+        name="zenml-vertex-quickstart",
+        model_name=current_model.name,
```

Comment:
will this always be the same name as the zenml mcp name?

Reply:
it's not forced but thinking about doing that and bypassing the user setting the name so we can always ensure it's only used when a model Is registered in MCP. I think Stefan also agrees on this